### ISSUE TITLE
Attempted to read from closed stream exception

### ISSUE THREAD

**Issue by mwiley63** [bytefish/FcmJava#18]
Excellent project!  I was able to integrate this library with a fresh Firebase project in less than 2 hours, and my Java backend was sending push notifications!

While the notification is still sent, on version 0.9, I get an exception when calling FCMClient.send() function (this exception does not appear on 0.8). 

Here is the code that triggers the exception (IFCMClientSettings & PersonData are copied from the readme):
```java
IFcmClientSettings clientSettings = new FixedFcmClientSettings("XXXXX");
// Create the Client using system-properties-based settings:
FcmClient client = new FcmClient(clientSettings);
// Message Options:
FcmMessageOptions options = FcmMessageOptions.builder().build();
NotificationPayload np = NotificationPayload.builder().setTitle("Hello world").setBody("AAbA").build();
// Send a Message:
List<String> ids = new ArrayList<String>();
ids.add("XXXXXX");
client.send(new DataMulticastMessage(options, ids, new PersonData("John", "Smith"), np));
```

Note, I get the same exception when using NotificationMutlicast and TopicMutlicast as well.

The relevant stack trace:
```
 Exception in thread "main" java.lang.RuntimeException: java.io.IOException: Attempted read from closed stream.
at de.bytefish.fcmjava.client.http.HttpClient.post(HttpClient.java:63)
at de.bytefish.fcmjava.client.FcmClient.post(FcmClient.java:99)
at de.bytefish.fcmjava.client.FcmClient.send(FcmClient.java:55)
```
A side question:  should I opt for 0.8 over 0.9?  What bugs were fixed with 0.9?  I could not find a changelog.

Thanks for the excellent code, cheers!

**Comment by bytefish** [comment]
I have introduced the Bug by adding a LoggingInterceptor for debugging mode. I consume the HTTP Request twice, which leads to the Exception. I will fix it and release a new version. Thanks for bringing up the issue. 👍

**Comment by bytefish** [comment]
A short summary on this bug, if someone finds this issue with Google. 

The problem was, that I was calling ``EntityUtils.toString`` twice on the entity to get the body from the request. So when the entity was accessed the second time, the Stream has already been consumed and the data was released. That's why the ``IOException`` was thrown. 

I think it is overkill to add Request Logging to the API, because it is already supported out of the box by Apache HttpComponents:

* [https://hc.apache.org/httpcomponents-client-ga/logging.html](https://hc.apache.org/httpcomponents-client-ga/logging.html) 

If Request and Response logging is neccessary, then the built-in Logging facilities should be used. 

I will release Version 1.0, which fixes this issue.

Thanks again for raising the issue.

**Comment by bytefish** [comment]
I have released version 1.0 to the Central Maven Repository. It should be synced in a few hours and the error should be fixed. Please reopen the ticket, if the fix doesn't work. :+1:
